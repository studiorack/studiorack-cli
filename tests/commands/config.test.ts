import path from 'path';
import { execa } from 'execa';
import { expect, test } from 'vitest';

const CLI: string = path.resolve('./', 'build', 'index.js');

test('Config get', async () => {
  const appDir = await execa('node', [CLI, 'config', 'get', 'appDir']);
  expect(appDir.stdout).toMatchSnapshot();
  const pluginsDir = await execa('node', [CLI, 'config', 'get', 'pluginsDir']);
  expect(pluginsDir.stdout).toMatchSnapshot();
  const presetsDir = await execa('node', [CLI, 'config', 'get', 'presetsDir']);
  expect(presetsDir.stdout).toMatchSnapshot();
  const projectsDir = await execa('node', [CLI, 'config', 'get', 'projectsDir']);
  expect(projectsDir.stdout).toMatchSnapshot();
  const registries = await execa('node', [CLI, 'config', 'get', 'registries']);
  expect(registries.stdout).toMatchSnapshot();
  const version = await execa('node', [CLI, 'config', 'get', 'version']);
  expect(version.stdout).toMatchSnapshot();
});

test('Config set', async () => {
  await execa('node', [CLI, 'config', 'set', 'appDir', 'test2']);
  const appDir = await execa('node', [CLI, 'config', 'get', 'appDir']);
  expect(appDir.stdout).toMatchSnapshot();

  await execa('node', [CLI, 'config', 'set', 'appDir', 'test']);
  const appDir2 = await execa('node', [CLI, 'config', 'get', 'appDir']);
  expect(appDir2.stdout).toMatchSnapshot();
});
