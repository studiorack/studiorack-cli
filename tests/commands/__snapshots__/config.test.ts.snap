// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Config get 1`] = `"test"`;

exports[`Config get 2`] = `"test/installed/plugins"`;

exports[`Config get 3`] = `"test/installed/presets"`;

exports[`Config get 4`] = `"test/installed/projects"`;

exports[`Config get 5`] = `
"[
  {
    name: 'Open Audio Registry',
    url: 'https://open-audio-stack.github.io/open-audio-stack-registry'
  }
]"
`;

exports[`Config get 6`] = `"1.0.0"`;

exports[`Config set 1`] = `"test2"`;

exports[`Config set 2`] = `"test"`;
